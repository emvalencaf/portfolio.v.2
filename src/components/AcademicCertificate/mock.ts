import { AcademicCertificateProps } from ".";

export default {
	courseTitle: "Curso de React.Js e Next.Js (Intermediário e Avançado)",
	institution: "Udemy, Instrutor: Luiz Otávio Miranda",
	courseWork: "93 horas",
	startIn: 2932983,
	endIn: 3932983,
	urlDownload: "/assets/files/UC-44225967-b7ec-4add-8983-426364ad7bb6.pdf"
} as AcademicCertificateProps;