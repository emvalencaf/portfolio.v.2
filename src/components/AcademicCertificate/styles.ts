import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;