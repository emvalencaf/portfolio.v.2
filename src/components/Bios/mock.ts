import { BiosProps } from ".";

export default {
	bios: `
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore rem minus dolore ducimus quos, sed dolores odit maiores quam tempora atque deserunt voluptas perferendis, aspernatur, sequi dignissimos reiciendis. In, esse!
	Lorem ipsum, dolor sit amet consectetur adipisicing elit. Non a deleniti quia. Magnam aspernatur cumque, nemo vero libero excepturi consequuntur corporis ullam nobis molestiae? Atque nam praesentium ullam quaerat quisquam?
</p>
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio cumque recusandae libero sequi, tempora iste, illo, possimus unde cum numquam quae corrupti minima tenetur mollitia earum repellendus animi molestiae voluptates.
	Lorem ipsum dolor sit, amet consectetur adipisicing elit. Obcaecati dolores fugiat ipsam ut odit sed aliquid ducimus molestiae vero ipsa nam veniam dolor libero labore qui pariatur, alias assumenda itaque?
	Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo cum temporibus nesciunt sequi odit ipsam harum magnam dolore a molestiae. Totam debitis, rerum aliquid sit non ratione. Et, omnis iusto!
</p>
<p>
	Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nam dicta aut cum sequi voluptatibus. Vel pariatur veritatis quidem. Culpa vero deleniti minima earum quod, odit deserunt fugit reprehenderit architecto in!
</p>
	`,
	profilePhoto: {
		altText: "profile picture",
		srcImg: "assets/images/potrait-photo.jpg",
	},
} as BiosProps;
