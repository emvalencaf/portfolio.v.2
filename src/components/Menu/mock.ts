import { MenuProps } from ".";

export default {
	menuLinks: [
		{
			children: "home",
			link: "/",
			newTab: false,
		},
		{
			children: "sobre",
			link: "/sobre",
			newTab: false,
		},
		{
			children: "habilidades",
			link: "/habilidades",
			newTab: false,
		},
		{
			children: "projetos",
			link: "/projetos",
			newTab: false,
		},
		{
			children: "curriculo",
			link: "/curriculo",
			newTab: false,
		}
	]
} as MenuProps;
