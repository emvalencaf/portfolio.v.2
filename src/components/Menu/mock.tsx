// icons
import {
	Home,
	Person3,
	Code,
	Cases,
	Portrait,
} from "@styled-icons/material-outlined";
// types
import { MenuProps } from ".";

export default {
	menuLinks: [
		{
			children: "home",
			link: "#",
			icon: <Home />,
			newTab: false,
		},
		{
			children: "sobre",
			link: "#sobre",
			icon: <Person3 />,
			newTab: false,
		},
		{
			children: "habilidades",
			icon: <Code />,
			link: "#habilidades",
			newTab: false,
		},
		{
			children: "projetos",
			icon: <Cases />,
			link: "#projetos",
			newTab: false,
		},
		{
			children: "curriculo",
			icon: <Portrait />,
			link: "#curriculo",
			newTab: false,
		},
	],
} as MenuProps;
