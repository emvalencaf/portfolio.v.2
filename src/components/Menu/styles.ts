import styled, { css } from 'styled-components';

export const Nav = styled.nav`
	${({ theme }) => css`
		@keyframes navbarLinkLoad {
    		0%{
        		transform: scale(0);
    		}

    		80%{
        		transform: scale(1.1);
    		}

    		100%{
        		transform: scale(1);
    		}
		}

		@keyframes navbarLinkLoad_first{
    		0%{
        		transform: translateX(-1000%);
    		}

    		50%{
       			transform: translateX(50%);
    		}
    		100%{
        		transform: translateX(0);
    		}
		}

		@keyframes navbarLinkLoad_last{
	    	0%{
    	    	transform: translateX(1000%);
    		}

    		50%{
        		transform: translateX(-50%);
    		}
    		100%{
        		transform: translateX(0);
    		}
		}

		& > ul {
			display: flex;
			justify-content: center;
			align-items: center;
			gap: ${theme.spacings.small};
		}

		& ul > li {
			list-style: none;
			animation: navbarLinkLoad;
			animation-duration: 1000ms;
			animation-delay: 450ms;
			animation-fill-mode: forwards;
			transform: scale(0);
		}

		& ul > li:first-child {
			animation-name: navbarLinkLoad_first;
    		transform: translateX(-1000%);
		}

		& ul > li:last-child {
			animation-name: navbarLinkLoad_last;
    		transform: translateX(1000%);
		}
	`}
`;

