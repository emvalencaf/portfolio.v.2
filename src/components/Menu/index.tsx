// styles
import Link from 'next/link';
import MenuLink, { MenuLinkProps } from '../MenuLink';
import * as Styled from './styles';

// types
export type MenuProps = {
	menuLinks?: MenuLinkProps[];
};

const Menu = ({ menuLinks = [] }: MenuProps) => {
	return (
		<Styled.Nav>
			<ul>
				{menuLinks.length >= 1 ? menuLinks.map((link) => (
					<li key={link.link}>
						<MenuLink link={link.link} newTab={link.newTab}>{link.children}</MenuLink>
					</li>
				)) : (
					<li>
						<MenuLink link="/">
							Home
						</MenuLink>
					</li>
				)}
			</ul>
		</Styled.Nav>
	);
};

export default Menu;
