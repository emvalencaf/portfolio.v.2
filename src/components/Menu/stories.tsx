import { Meta, Story } from '@storybook/react/types-6-0';
import Menu, { MenuProps } from '.';
import mock from './mock';

export default {
	title: 'Menu',
	component: Menu,
	args: mock,
	parameters: {
		layout: "fullscreen",
	},
} as Meta<MenuProps>;

export const Template: Story<MenuProps> = (args) => {
	return (
		<div>
			<Menu {...args} />
		</div>
	);
};
