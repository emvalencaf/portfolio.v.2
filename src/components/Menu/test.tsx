import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

// react component to be test
import Menu from '.';

// mock
import mock from './mock';

describe('<Menu />', () => {
	it("should render a Menu component with mock MenuLinks", () => {
		renderTheme(<Menu {...mock} />);

		expect(screen.getAllByRole("link")).toHaveLength(mock.menuLinks.length);
	});

	it("should render a Menu component without any props", () => {
		const { container } = renderTheme(<Menu />);

		expect(screen.getByRole("link")).toHaveAttribute("href", "/");
	});

	it("should match a snapshot", () => {
		const { container } = renderTheme(<Menu {...mock} />);

		expect(container).toMatchSnapshot();
	})
});
