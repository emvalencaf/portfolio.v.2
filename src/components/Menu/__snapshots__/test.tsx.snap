// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match a snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #0A1128;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 350ms ease-in;
  transition: all 350ms ease-in;
}

.c1:hover {
  color: #dc143c;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c0 > ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
}

.c0 ul > li {
  list-style: none;
  -webkit-animation: navbarLinkLoad;
  animation: navbarLinkLoad;
  -webkit-animation-duration: 1000ms;
  animation-duration: 1000ms;
  -webkit-animation-delay: 450ms;
  animation-delay: 450ms;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c0 ul > li:first-child {
  -webkit-animation-name: navbarLinkLoad_first;
  animation-name: navbarLinkLoad_first;
  -webkit-transform: translateX(-1000%);
  -ms-transform: translateX(-1000%);
  transform: translateX(-1000%);
}

.c0 ul > li:last-child {
  -webkit-animation-name: navbarLinkLoad_last;
  animation-name: navbarLinkLoad_last;
  -webkit-transform: translateX(1000%);
  -ms-transform: translateX(1000%);
  transform: translateX(1000%);
}

<div>
  <nav
    class="c0"
  >
    <ul>
      <li>
        <a
          class="c1"
          href="/"
          rel="external"
          target="_self"
        >
          home
        </a>
      </li>
      <li>
        <a
          class="c1"
          href="/sobre"
          rel="external"
          target="_self"
        >
          sobre
        </a>
      </li>
      <li>
        <a
          class="c1"
          href="/habilidades"
          rel="external"
          target="_self"
        >
          habilidades
        </a>
      </li>
      <li>
        <a
          class="c1"
          href="/projetos"
          rel="external"
          target="_self"
        >
          projetos
        </a>
      </li>
      <li>
        <a
          class="c1"
          href="/curriculo"
          rel="external"
          target="_self"
        >
          curriculo
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
